inst/templates/ad_advs.R

```diff
@@ -0,0 +1,552 @@
+
+# Get list of ADSL vars required for derivations
+adsl_vars <- exprs(SEX, BRTHDTC, TRTSDT, TRTEDT, TRT01A, TRT01P)
+
+advs <- vs %>%
+  # Join ADSL with VS (need BRTHDT for AAGECUR derivation)
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = exprs(!!!get_admiral_option("subject_keys"))
+  ) %>%
+  mutate(BRTHDT = convert_dtc_to_dt(
+    BRTHDTC,
+    highest_imputation = "n",
+    date_imputation = "first",
+    min_dates = NULL,
+    max_dates = NULL,
+    preserve = FALSE
+  )) %>%
+  ## Calculate ADT, ADY ----
+  derive_vars_dt(
+    new_vars_prefix = "A",
+    dtc = VSDTC
+  ) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT)) %>%
+  ## Calculate Current Analysis Age AAGECUR ----
+  derive_vars_duration(
+    new_var = AAGECUR,
+    new_var_unit = AAGECURU,
+    start_date = BRTHDT,
+    end_date = ADT,
+    out_unit = "DAYS",
+    trunc_out = FALSE
+  )
+
+advs <- advs %>%
+  ## Add PARAMCD only - add PARAM etc later ----
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup %>% filter(!is.na(VSTESTCD)),
+    new_vars = exprs(PARAMCD),
+    by_vars = exprs(VSTESTCD)
+  ) %>%
+  ## Calculate AVAL ----
+  mutate(AVAL = VSSTRESN)
+
+## Get visit info ----
+# See also the "Visit and Period Variables" vignette
+# (https://pharmaverse.github.io/admiral/articles/visits_periods.html#visits)
+advs <- advs %>%
+  # Derive Timing
+  mutate(
+    ATPTN = VSTPTNUM,
+    ATPT = VSTPT,
+    AVISIT = case_when(
+      str_detect(VISIT, "UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      !is.na(VISIT) ~ str_to_title(VISIT),
+      TRUE ~ NA_character_
+    ),
+    AVISITN = as.numeric(case_when(
+      VISIT == "SCREENING 1" ~ "-1",
+      VISIT == "BASELINE" ~ "0",
+      str_detect(VISIT, "WEEK") ~ str_trim(str_replace(VISIT, "WEEK", "")),
+      TRUE ~ NA_character_
+    ))
+  )
+
+# Add Current HEIGHT Temporary variable (in cm)
+## Calculate Current HEIGHT/LENGTH at each time point Temporary variable----
+hgttmp <- advs %>%
+  filter((PARAMCD == "HEIGHT" | PARAMCD == "LENGTH") & VSSTRESU == "cm") %>%
+  select(STUDYID, USUBJID, HGTTMP = AVAL, HGTTMPU = VSSTRESU, AVISIT)
+
+advs <- advs %>%
+  derive_vars_merged(
+    dataset_add = hgttmp,
+    by_vars = exprs(STUDYID, USUBJID, AVISIT)
+  )
+
+## Merge ADVS to the chosen Growth metadata as an input to meta_criteria ----
+## Calculate z-scores/percentiles
+## Calculate Weight for AGE z-score and Percentile ----
+## Note: PARAMN needs to be updated.
+advs_age <- advs %>% derive_params_growth_age(
+  sex = SEX,
+  age = AAGECUR,
+  age_unit = AAGECURU,
+  meta_criteria = weight_for_age,
+  parameter = VSTESTCD == "WEIGHT",
+  analysis_var = AVAL,
+  set_values_to_sds = exprs(
+    PARAMCD = "WTASDS",
+    PARAM = "Weight-for-age z-score"
+  ),
+  set_values_to_pctl = exprs(
+    PARAMCD = "WTAPCTL",
+    PARAM = "Weight-for-age percentile"
+  )
+)
+
+## Calculate BMI for AGE z-score and Percentile ----
```

Comment:
Add a section title such as "Derive Anthropometric indicators parameters" above this section. I see Zelos has been using that wording in his function headers